src/button/AriaToggleButton.ts

```diff
@@ -0,0 +1,40 @@
+import * as React from "react";
+import { useForkRef } from "reakit-utils";
+import { BoxHTMLProps, useBox } from "reakit";
+import { mergeProps } from "@react-aria/utils";
+import { ToggleState } from "@react-stately/toggle";
+import { useToggleButton } from "@react-aria/button";
+import { INTERACTION_KEYS } from "../interactions/__keys";
+import { createComponent, createHook } from "reakit-system";
+import { AriaToggleButtonProps } from "@react-types/button";
+
+type useAriaToggleButtonOptions = AriaToggleButtonProps & ToggleState;
+
+export const useAriaToggleButton = createHook<
+  useAriaToggleButtonOptions,
+  BoxHTMLProps
+>({
+  name: "useAriaToggleButton",
+  keys: [...INTERACTION_KEYS, "isSelected", "setSelected", "toggle"],
+  compose: [useBox],
+
+  useProps(options, { ref: htmlRef, ...htmlProps }) {
+    const { isSelected, setSelected, toggle, ...restOptions } = options;
+    const props = { ...restOptions, ...htmlProps } as AriaToggleButtonProps;
+    const state = { isSelected, setSelected, toggle } as ToggleState;
```

Comment:
Is there any particular reason why we are using `as` assertion instead of using type declarations directly?

```ts
    const props: AriaToggleButtonProps = { ...restOptions, ...htmlProps };
    const state: ToggleState = {
      isSelected,
      setSelected,
      toggle,
    };
```

I can also see that using `const props: AriaToggleButtonProps = { ...restOptions, ...htmlProps };` typescript is complaining about some mismatched types.


<details>
<summary>Ts error</summary>

```ts
const props: AriaToggleButtonProps
Type '{ defaultChecked?: boolean | undefined; defaultValue?: string | number | readonly string[] | undefined; suppressContentEditableWarning?: boolean | undefined; suppressHydrationWarning?: boolean | undefined; ... 263 more ...; excludeFromTabOrder?: boolean | undefined; }' is not assignable to type 'AriaToggleButtonProps'.
  Types of property 'onChange' are incompatible.
    Type '((event: FormEvent<any>) => void) | ((isSelected: boolean) => void) | undefined' is not assignable to type '((isSelected: boolean) => void) | undefined'.
      Type '(event: FormEvent<any>) => void' is not assignable to type '(isSelected: boolean) => void'.
        Types of parameters 'event' and 'isSelected' are incompatible.
          Type 'boolean' is not assignable to type 'FormEvent<any>'.ts(2322)
```

</details>

Reply:
That is because the `useToggleButton` only accepts props of type `AriaToggleButtonProps`.

But `htmlPorps` has more types that are not accepted by `AriaToggleButtonProps`, so I had to override them.

`const props: AriaToggleButtonProps = { ...restOptions, ...htmlProps };`

By using directly, it doesn't match with the spreaded props, so had to use `as` to override it.

Reply:
Hmmm i see okay, i guess it's fine to use as assertion for this case.